app/models/department.rb

```diff
@@ -1,4 +1,22 @@
 class Department < ApplicationRecord
   validates_presence_of :name
   has_many :participants
+
+  def self.number_of_participants
+    departments = Department.all
+    departments.each do |department|
+      department.number_of_participants = department.participants.count
+      department.save
+    end
+  end
+
+  def self.calculate_average_score
+    departments = Department.all
+    departments.each do |department|
+      individual_scores = department.participants.pluck(:fika_score)
+      num_participants = department.participants.count
```

Comment:
Would `department` not have the `number_of_participants` at this point? If that function is always run first in the controller, we might be able to delete this line

Reply:
It would indeed